boards/arm/mimxrt1024_evk/mimxrt1024_evk_defconfig

```diff
@@ -14,3 +14,4 @@ CONFIG_GPIO=y
 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=500000000
 CONFIG_ARM_MPU=y
 CONFIG_HW_STACK_PROTECTION=y
+CONFIG_FLASH_MCUX_FLEXSPI_NOR=y
```

Comment:
Why are we moving this to the defconfig?

Reply:
It does not work having it in Kconfig.defconfig when the choice is optional. It does not get chosen.

Reply:
It will get selected when `CONFIG_FLASH` is defined in the user application.

Reply:
> It will get selected when CONFIG_FLASH is defined in the user application.

@mmahadevan108  correct if this was a regular choice:
https://github.com/zephyrproject-rtos/zephyr/blob/11fc938e9163d6dc1d25a468f25cceca7e29d8d1/drivers/flash/Kconfig.mcux#L30

but it's not, because it has the `optional` keyword.
https://github.com/zephyrproject-rtos/zephyr/blob/11fc938e9163d6dc1d25a468f25cceca7e29d8d1/drivers/flash/Kconfig.mcux#L30-L32

So if doing as you propose, the `FLASH_MCUX_FLEXSPI_NOR` will correctly be the default value, but the whole choice group will still be disabled, like in this pic:
![image](https://user-images.githubusercontent.com/9802583/134737866-605c22b2-bfd7-42b3-b226-6fa65e5672d3.png)

The optional choice group `FLEXSPI_FLASH_DRIVER` cannot be enabled using the `Kconfig.defconfig`, but if you select the specific `CONFIG_FLASH_MCUX_FLEXSPI_NOR=y` in the `_defconfig` file, then the choice group will be enabled and `FLASH_MCUX_FLEXSPI_NOR` will be the selected flash driver.

Reply:
I would suggest removing the `optional` keyword. If the Flash driver is selected then we would need one of the drivers.

Reply:
it does seem odd to be moving this here instead of leaving it where it was. I understand that there were unknown problems with the previous arrangement but seems like we should figure out what was the issue/dependency with the original setup.

Reply:
The reason for this change was:
Making the FLEXSPI_FLASH_DRIVER optional meant we could not choose it in the board Kconfig.defconfig. 
This made flash and fs tests fail in buildkite.

Reply:
> I would suggest removing the optional keyword. 

@dleach02  The existing solution on master has this:
https://github.com/zephyrproject-rtos/zephyr/blob/921a3d8543429c5b58a91c115d8c1cc416cb569a/drivers/flash/Kconfig.mcux#L30-L31
https://github.com/zephyrproject-rtos/zephyr/blob/921a3d8543429c5b58a91c115d8c1cc416cb569a/drivers/flash/Kconfig.mcux#L37-L38

meaning that a user using `zephyr/main` can do:
- Disable all drivers
- Select multiple drivers (definitely an error)

Now, the choice ensures that only a single driver can be selected.
The `optional` ensures that the existing possibility of deselecting all drivers is still possible.

You may argue that deselecting all drivers is already an error in `zephyr/main` but are we sure that there are no users relying on the fact that they may deselect all driver ? (for example if providing their own implementation)

Reply:
@tejlmand, I agree that the existing solution is an error in that you can select multiple drivers so I do want this to change to a 'choice'. See my comment at the bottom of the conversation thread about how you can retain the deselection of all drivers and avoid touching the <board>_defconfig files.